### ISSUE TITLE
Solved: iOS mobile Safari and Firefox not receiving clicks

### ISSUE THREAD

**Issue by jollygreenlaser** [leptos-rs/leptos#2381]
Turns out iOS goes against every other browser out there and does not send click events on touches unless it's a button or a. Unless! You put onclick="" on any element above it in the tree. Not Leptos specific but definitely something you'll hit developing a website with Leptos. 

That is, things that use `on:click` and work fine in Android or Chrome iOS, will only fire `on:touch` by default in iOS Safari and Firefox. 

Hopefully this helps someone else struggling with why it seems like wasm is broken or clicks aren't doing anything on mobile safari or iOS firefox.

Putting this up and closing it immediately as a solved issue that someone else may find via search in the future.

**Comment by mondeja** [comment]
See https://stackoverflow.com/a/39712411 and [how leptos-use handles this](https://github.com/Synphonyte/leptos-use/blob/7216a3281c95cfbfa1eea4de4a26face1a4ee66f/src/on_click_outside.rs#L124C26-L143).

**Comment by jollygreenlaser** [comment]
@mondeja Thanks! That stackoverflow was one of the resources I used to debug this. 

I had originally encountered this issue on iOS Safari and found that `on_click_outside` used outside my `Router` (in `app.rs`) fixed this. It seems that `leptos-use` fix is why. However, I didn't realize the issue persisted in iOS Firefox. I needed to go all the way to adding the explicit `onclick=""` to fix it there.
